src/visualize/image_like.jl

```diff
@@ -127,35 +109,114 @@ function play{T}(buffer::Array{T, 2}, video_stream, t)
     return reinterpret(T, buffer, (w,h))
 end
 
-"""
-Takes a 3D image and decides if it is a volume or an animated Image.
-"""
-function _default{T<:Colorant, X}(img::Images.Image{T, 3, X}, s::Style, data::Dict)
-    props = img.properties
-    if haskey(props, "timedim")
-        if haskey(props, "spatialorder")
-            get!(data, :spatialorder, join(props["spatialorder"], ""))
+# If the user is using the new Images, ImageAxes will be loaded
+if isdefined(Images, :ImageAxes)
+    unwrap(img::Images.ImageMeta) = unwrap(data(img))
```

Comment:
You can do
```julia
foo{A<:AbstractArray}(x::Signal{A}) = ...
```
Or do you mean the lack of triangular dispatch? If you have to, one strategy is to create two arguments from a single argument:
```julia
foo{C<:Colorant}(A::AbstractArray{C}) = _foo(A, eltype(C))
_foo{C<:Colorant,T<:FixedPoint}(A::AbstractArray{C}, ::Type{T}) = ...
```
A somewhat easier way is to use a typealias:
```julia
@compat const FixedColorant{T<:FixedPoint,N} = Colorant{T,N}
@compat const AbstractColorantFixedArray{C<:FixedColorant,N} = AbstractArray{C,N}
```
Though in general I try to avoid dispatching on element type when possible (but sometimes you simply can't avoid it).